src/fs/metadata_log.cc

```diff
@@ -266,6 +279,10 @@ metadata_log::flush_result metadata_log::schedule_flush_of_curr_cluster_and_chan
 
 void metadata_log::schedule_attempt_to_delete_inode(inode_t inode) {
     return schedule_background_task([this, inode] {
+        if (_read_only) {
+            return make_exception_future(read_only_filesystem_exception());
+        }
```

Comment:
please move this `if` to a helper function (`metadata_log::throw_if_read_only()` or something like that) in order to reduce the number of lines we need to type every time we want to check this condition and throw.